### PR TITLE
Add examples for transactions with the different signer types

### LINKED ISSUE #240: Add examples for transactions with the different signer types
ℹ️ This issue is part of Epic #237 

ℹ️ Related to #123 as well 

## Objective

- Add examples for transactions with the different signer types.

## Checklist

- [ ] Perform a payment with `ed25519` signer
- [ ] Perform a payment with `sha256_hash` signer
- [ ] Perform a payment with `pre_auth_tx` signer
- [ ] Perform a payment with `signed_payload` signer

### PR DESCRIPTION
**Pull request by miguelnietoa**
Closes #240

### CODE REVIEW

File: docs/examples.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
 # Examples
 * [Creating an account](/docs/examples/create_account.md)
 * [Payments](/docs/examples/payments.md)
+* [Hash(x) Signer](/docs/examples/hash_x.md)
+* [Pre-authorized Transaction Signer](/docs/examples/pre_auth_tx.md)
+* [Ed25519 Signed Payload Signer](/docs/examples/ed25519_signed_payload.md)
```
Comment (juanhurtado10) [inline docs/examples.md:6]: @miguelnietoa let's replace signer by signature

File: docs/examples/pre_auth_tx.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+# 9. build the set options operation setting the trasaction hash as a signer with weight 1
+set_options_op =
+  Stellar.TxBuild.SetOptions.new(
+    signer: [
+      pre_auth_tx: payment_tx_hash,
+      weight: 1
+    ]
+  )
+
+# 10. build the signature for the set options transaction
```
Comment (juanhurtado10) [inline docs/examples/pre_auth_tx.md:61]: @miguelnietoa if you are pre-authorizing a transaction why is this signature necessary?

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

File: docs/examples.md (hunk 1 above)
Comment (miguelnietoa) [reply docs/examples.md]: done!

File: docs/examples/pre_auth_tx.md (hunk 2 above)
Comment (miguelnietoa) [reply docs/examples/pre_auth_tx.md:61]: This is the transaction that sets the signer of type `pre_auth_tx` and it should be signed.

But the payment transaction **should not be signed** as it is actually a pre-authorized transaction.

**Review by juanhurtado10 — APPROVED** [review]

**Pull request merged**